src/Query/QueryInterface.php

```diff
@@ -306,4 +306,9 @@ public function scalar(): bool|int|null|string|float;
      * @return bool Whether to emulate query execution.
      */
     public function shouldEmulateExecution(): bool;
+
+    /**
+     * Enables or disables typecasting of values when retrieving records.
+     */
+    public function typecasting(bool $typecasting = true): static;
```

Comment:
```suggestion
    public function phpTypecasting(bool $typecasting = true): static;
```

For consistency with `CommandInterface`.

Reply:
`QueryInterface` has only `db -> php` typecasting
as in another PR #949
`DMLQueryBuilderInterface` has only `php -> db` typecasting (immutable `withTypecasting()`)

For simplicity, the name `typecasting()` looks better since there are no other options.

`CommandInterface` has two typecasting options `db -> php` and `php -> db`
After merging the PRs I plan to add `CommandInterface::typecasting()` to change both options.

Also, do you have any thoughts on the `DMLQueryBuilderInterface::withTypecasting()` method in #949 PR?
This is not consistent with the `typecasting()` naming, but it should be immutable.